pkg/utils/pod.go

```diff
@@ -207,11 +207,27 @@ func maxResourceList(list, new v1.ResourceList) {
 
 // PodToleratesTaints returns true if a pod tolerates one node's taints
 func PodToleratesTaints(pod *v1.Pod, taintsOfNodes map[string][]v1.Taint) bool {
+
 	for nodeName, taintsForNode := range taintsOfNodes {
-		if len(pod.Spec.Tolerations) >= len(taintsForNode) && TolerationsTolerateTaintsWithFilter(pod.Spec.Tolerations, taintsForNode, nil) {
-			return true
+		if len(pod.Spec.Tolerations) >= len(taintsForNode) {
+
+			taint, ok := TolerationsTolerateTaints(pod.Spec.Tolerations, taintsForNode)
```

Comment:
I'm sorry @ingvagabund I do not feel your change is simpler/clearer. It feels less idiomatic to maintain a function just for debug, and testing for the verbosity level is useless as the log will only be displayed if level > 5 anyways.

This is my first PR in the project, so please let me know what you think and resolve the conversation if you fell it's OK the way I submitted it. If not i'll change it.

Reply:
In that case I misunderstood purpose of your PR. I assumed goal of this PR is to print the conflicting taints when log level 5 or higher is set. Is your PR still WIP or ready for review?

>  It feels less idiomatic to maintain a function just for debug,

There's no new function introduced in my suggestion.

> and testing for the verbosity level is useless as the log will only be displayed if level > 5 anyways.

That was my observation from this PR. If that is not your intention, please be more explicit about it.

> This is my first PR in the project, so please let me know what you think and resolve the conversation if you fell it's OK the way I submitted it. If not i'll change it.

What is the goal here then? Would you more elaborate?

Reply:
The PR is just to add more informative logs when:
- there is a Toleration missmatch preventing the pod to be scheduled on a node
- the loglevel is > 5 (because this info is not mandatory for regular usage)

For the last point, we could send the info back to the previous function call, which is logging at level 3 (default loglevel), but it would be a bigger change for information detail we don't need.

I think I missed the inner for-loop in your proposal too. I get the idea now... so yeah, I will refactor the PR as you suggested.
thanks

Reply:
@ingvagabund I made the requested changes. Thanks for your help and answers